From: Rahul Rameshbabu <rrameshbabu@nvidia.com>
Date: Fri, 24 Oct 2025 19:30:02 -0700
Subject: tests/monitor-abstraction: Assign non-conflicting CRTCs for tiled
 monitors

The preferred-mode and changed-connector tests would have conflicting CRTCs
between the normal monitor and output for the primary tile of the tiled
monitor in the tests. This change removes the conflict to prevent failures
in the meta_monitor_config_manager_assign path.

Author: Rahul Rameshbabu <rrameshbabu@nvidia.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2131575
Origin: upstream, https://gitlab.gnome.org/GNOME/mutter/-/commit/9be70f1723ac9b8c884930afbde2bbd9e930f1e3
Forwarded: yes
Reviewed-By: Mitchell Augustin <mitchell.augustin@canonical.com>
Last-Update: 2025-11-19

Signed-off-by: Rahul Rameshbabu <rrameshbabu@nvidia.com>
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4685>
(cherry picked from commit 9be70f1723ac9b8c884930afbde2bbd9e930f1e3)
Signed-off-by: Mitchell Augustin <mitchell.augustin@canonical.com>
---
 src/tests/monitor-abstraction-tests.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/tests/monitor-abstraction-tests.c b/src/tests/monitor-abstraction-tests.c
index ed06615..8a818e3 100644
--- a/src/tests/monitor-abstraction-tests.c
+++ b/src/tests/monitor-abstraction-tests.c
@@ -720,7 +720,7 @@ meta_test_monitor_rebuild_preferred_mode (void)
         .serial = "0x10000",
       },
       {
-        .crtc = 0,
+        .crtc = 1,
         .modes = { 1, 3 },
         .n_modes = 2,
         .preferred_mode = 1,
@@ -740,7 +740,7 @@ meta_test_monitor_rebuild_preferred_mode (void)
         }
       },
       {
-        .crtc = 1,
+        .crtc = 2,
         .modes = { 1, 3 },
         .n_modes = 2,
         .preferred_mode = 0,
@@ -851,7 +851,7 @@ meta_test_monitor_rebuild_changed_connector (void)
         .connector_number = 1,
       },
       {
-        .crtc = 0,
+        .crtc = 1,
         .modes = { 1, },
         .n_modes = 1,
         .preferred_mode = 1,
@@ -872,7 +872,7 @@ meta_test_monitor_rebuild_changed_connector (void)
         .connector_number = 2,
       },
       {
-        .crtc = 1,
+        .crtc = 2,
         .modes = { 1, },
         .n_modes = 1,
         .preferred_mode = 0,
